uploader/push_to_sling.py

```diff
@@ -4,28 +4,60 @@
 import argparse
 import getpass
 import logging
-from pathlib import Path
+import yaml
 from pathlib import PurePath
 from datetime import datetime
 
+DEFAULT_SERVER = 'http://localhost:8080'
+DEFAULT_REPOSITORY = getpass.getuser() + '_' + datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+DEFAULT_USER = 'demo'
+DEFAULT_PASSWORD = 'demo'
+DEFAULT_LINKS = False
+
 HEADERS = {'cache-control': 'no-cache',
            'Accept': 'application/json'}
 
+
+def matches(path, globs, globType):
+    for glob in globs:
+        if path.match(glob):
+            logger.debug('File %s matches on %s glob %s', file, globType, glob)
+            return True
+    return False
+
+
 parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description='''\
 Red Hat bulk upload module for Pantheon 2. This tool will scan a directory recursively and upload relevant files.
 
 Both this uploader and Pantheon 2 are ALPHA software and features may update or change over time. 
  
 ''')
-parser.add_argument('--server', '-s', help='The Pantheon server to upload modules to, default http://localhost:8080', default='http://localhost:8080')
-parser.add_argument('--repository', '-r', help='The name of the repository; this is a convenience for the user. If not specified, one will be autogenerated', default=getpass.getuser() + '_' + datetime.now().strftime('%Y-%m-%d_%H-%M-%S'))
-parser.add_argument('--user', '-u', help='Username for authentication, default \'demo\'', default='demo')
-parser.add_argument('--password', '-p', help='Password for authentication, default \'demo\'. If \'-\' is supplied, the script will prompt for the password.', default='demo')
+parser.add_argument('--server', '-s', help='The Pantheon server to upload modules to, default ' + DEFAULT_SERVER)
+parser.add_argument('--repository', '-r', help='The name of the repository; this is a convenience for the user. If not specified, one will be autogenerated')
+parser.add_argument('--user', '-u', help='Username for authentication, default \'' + DEFAULT_USER + '\'', default=DEFAULT_USER)
+parser.add_argument('--password', '-p', help='Password for authentication, default \'' + DEFAULT_PASSWORD + '\'. If \'-\' is supplied, the script will prompt for the password.', default=DEFAULT_PASSWORD)
```

Comment:
Consider removing the default username/password combination. Once in production, users will need to specify credentials every time, so might as well enforce the requirement early on.

Reply:
Yeah, we should have a conversation about credentials. I would be open to storing them in a separate config file, maybe ~/.config/pantheon2.cfg, and we check that the file is chmod 600 or else we throw an error, similar to ssh. As an advanced feature, we could also assign per-user tokens/keys, although I'm not sure there's enough payoff there, but it's a possibility.

For now, since the demo credentials are the only ones that are available, I think leaving them here as defaults makes sense. When we get real auth working, let's revisit this.

Reply:
:+1: